spec/turbot/auth_spec.rb

```diff
@@ -141,74 +141,74 @@ module Turbot
-            @cli.should_receive(:ask).and_return("2")
-            @cli.check_for_associated_ssh_key
-          end
-        end
-      end
-    end
+    # describe "automatic key uploading" do
+    #   before(:each) do
+    #     FileUtils.mkdir_p("#{@cli.home_directory}/.ssh")
+    #     @cli.stub!(:ask_for_credentials).and_return("username", "apikey")
+    #   end
+
+    #   describe "an account with existing keys" do
+    #     before :each do
+    #       @api = mock(Object)
+    #       @response = mock(Object)
+    #       @response.should_receive(:body).and_return(['existingkeys'])
+    #       @api.should_receive(:get_ssh_keys).and_return(@response)
+    #       @cli.should_receive(:api).and_return(@api)
+    #     end
+
+    #     it "should not do anything if the account already has keys" do
+    #       @cli.should_not_receive(:associate_key)
+    #       @cli.check_for_associated_ssh_key
+    #     end
+    #   end
+
+    #   describe "an account with no keys" do
+    #     before :each do
+    #       @api = mock(Object)
+    #       @response = mock(Object)
+    #       @response.should_receive(:body).and_return([])
+    #       @api.should_receive(:get_ssh_keys).and_return(@response)
+    #       @cli.should_receive(:api).and_return(@api)
+    #     end
+
+    #     describe "with zero public keys" do
+    #       it "should ask to generate a key" do
+    #         @cli.should_receive(:ask).and_return("y")
+    #         @cli.should_receive(:generate_ssh_key).with("id_rsa")
+    #         @cli.should_receive(:associate_key).with("#{@cli.home_directory}/.ssh/id_rsa.pub")
+    #         @cli.check_for_associated_ssh_key
+    #       end
+    #     end
+
+    #     describe "with one public key" do
+    #       before(:each) { FileUtils.touch("#{@cli.home_directory}/.ssh/id_rsa.pub") }
+    #       after(:each)  { FileUtils.rm("#{@cli.home_directory}/.ssh/id_rsa.pub") }
+
+    #       it "should upload the key" do
+    #         @cli.should_receive(:associate_key).with("#{@cli.home_directory}/.ssh/id_rsa.pub")
+    #         @cli.check_for_associated_ssh_key
+    #       end
+    #     end
+
+    #     describe "with many public keys" do
+    #       before(:each) do
+    #         FileUtils.touch("#{@cli.home_directory}/.ssh/id_rsa.pub")
+    #         FileUtils.touch("#{@cli.home_directory}/.ssh/id_rsa2.pub")
+    #       end
+
+    #       after(:each) do
+    #         FileUtils.rm("#{@cli.home_directory}/.ssh/id_rsa.pub")
+    #         FileUtils.rm("#{@cli.home_directory}/.ssh/id_rsa2.pub")
+    #       end
+
+    #       it "should ask which key to upload" do
+    #         File.open("#{@cli.home_directory}/.ssh/id_rsa.pub", "w") { |f| f.puts }
+    #         @cli.should_receive(:associate_key).with("#{@cli.home_directory}/.ssh/id_rsa2.pub")
+    #         @cli.should_receive(:ask).and_return("2")
+    #         @cli.check_for_associated_ssh_key
+    #       end
+    #     end
+    #   end
+    # end
   end
 end
```

Comment:
I used comments, because you used comments for this line: https://github.com/openc/turbot-client/blob/master/lib/turbot/auth.rb#L209

Presumably, if you were to uncomment that line, we would want to uncomment the associated tests.

That said, I prefer deleting lines to commenting out lines in all cases, and I can delete all commented-out code if that's preferred (there's a bunch, e.g. `bots:single`).